### PR TITLE
use 30 secs timeout for put vmagent log api

### PR DESCRIPTION
**Pull request by nagworld9**
## Description
exercise 30 secs timeout to reduce latency failure rates on PUT VMagent log api.
Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/utils/restutil.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def redact_sas_tokens_in_urls(url):
 
 
 def _http_request(method, host, rel_uri, port=None, data=None, secure=False,
-                  headers=None, proxy_host=None, proxy_port=None, redact_data=False):
+                  headers=None, proxy_host=None, proxy_port=None, redact_data=False, timeout=10):
```
Comment (narrieta) [inline azurelinuxagent/common/utils/restutil.py:313]: We defining the default timeout in multiple places. We should probably do that on the higher level functions (e.g. http_put) and make a mandatory parameter in the other methods (http_request, _http_request)

**Review by narrieta — COMMENTED** [review]
1 comment

File: azurelinuxagent/common/utils/restutil.py (hunk 1 above)
Comment (nagworld9) [reply azurelinuxagent/common/utils/restutil.py:313]: That would force us to change every http_* API call (GET, POST, etc). I'm okay to do that too.

Comment (narrieta) [reply azurelinuxagent/common/utils/restutil.py:313]: yes, I think that is a better API interface (which requires more code changes, but hey...) :)

Comment (narrieta) [reply azurelinuxagent/common/utils/restutil.py:313]: (Unless you think otherwise)

Comment (nagworld9) [reply azurelinuxagent/common/utils/restutil.py]: Sometimes better things come with price. I'm gonna pay for it if things are clean :)

**Review by narrieta — COMMENTED** [review]
thanks for the update

**Review by narrieta — APPROVED** [review]

**Pull request merged**